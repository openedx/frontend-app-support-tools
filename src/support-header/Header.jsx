/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/anchor-has-content */
import React, { useContext } from 'react';
import Responsive from 'react-responsive';
import { AppContext } from '@edx/frontend-platform/react';
import {
  APP_CONFIG_INITIALIZED,
  ensureConfig,
  mergeConfig,
  getConfig,
  subscribe,
} from '@edx/frontend-platform';

import DesktopHeader from './DesktopHeader';
import MobileHeader from './MobileHeader';

import LogoSVG from './logo.svg';

ensureConfig([
  'LMS_BASE_URL',
  'LOGOUT_URL',
  'LOGIN_URL',
  'MARKETING_SITE_BASE_URL',
  'ORDER_HISTORY_URL',
], 'Header component');

subscribe(APP_CONFIG_INITIALIZED, () => {
  mergeConfig({
    MINIMAL_HEADER: !!process.env.MINIMAL_HEADER,
    PUBLISHER_BASE_URL: process.env.PUBLISHER_BASE_URL,
  }, 'Header additional config');
});

export default function Header() {
  const { authenticatedUser, config } = useContext(AppContext);

  const mainMenu = [
    {
      type: 'submenu',
      content: 'Updates',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href="https://docs.google.com/document/edit?hgd=1&id=17_FWEtnUXyeEMrj8Lu3kmXqopwHGUjKW2t61cek2ax0">Current Issues</a></div>
          <div className="mb-1"><a rel="noopener" href="https://sites.google.com/a/edx.org/edx-support/-recent-updates">Support Site Recent Updates</a></div>
        </>
      ),
    },
    {
      type: 'submenu',
      content: 'Escalations',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href="https://edx.lightning.force.com/lightning/o/Case/list?filterName=00B0L000005D8BgUAK">Salesforce</a></div>
          <div className="mb-1"><a rel="noopener" href="https://openedx.atlassian.net/servicedesk/customer/user/requests?page=1&status=open">CR Requests</a></div>
        </>
      ),
    },
    {
      type: 'submenu',
      content: 'Payment',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.ECOMMERCE_BASE_URL}/dashboard/users/`}>Otto</a></div>
          <div className="mb-1"><a rel="noopener" href="https://ebc2.cybersource.com/ebc2/app/Home">Cybersource</a></div>
          <div className="mb-1"><a rel="noopener" href="https://www.paypal.com/mep/dashboard">Paypal</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.ECOMMERCE_BASE_URL}/enterprise/coupons/`}>Enterprise Coupons</a></div>
        </>
      ),
    },
    {
      type: 'submenu',
      content: 'Courses',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.PUBLISHER_BASE_URL}`}>Publisher</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.DISCOVERY_API_BASE_URL}`}>Discovery</a></div>
          <div className="mb-1"><a rel="noopener" href="https://www.edx.org/course">Course Catalogue</a></div>
        </>
      ),
    },
    {
      type: 'item',
      content: 'Proctoring',
      href: 'https://courses-internal.edx.org/admin/edx_proctoring/proctoredexamsoftwaresecurereview/',
    },
    {
      type: 'submenu',
      content: 'Programs',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.CREDENTIALS_BASE_URL}/records/`}>Learner Record</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.CREDENTIALS_BASE_URL}/admin/credentials/usercredential/`}>Credentials Search</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.DISCOVERY_API_BASE_URL}/admin/course_metadata/program/`}>Programs Discovery</a></div>
        </>
      ),
    },
  ];

  const dashboardMenuItem = {
    type: 'item',
    href: `${config.LMS_BASE_URL}/dashboard`,
    content: 'Dashboard',
  };

  const logoutMenuItem = {
    type: 'item',
    href: config.LOGOUT_URL,
    content: 'Logout',
  };

  let userMenu = authenticatedUser === null ? [] : [
    dashboardMenuItem,
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/u/${authenticatedUser.username}`,
      content: 'Profile',
    },
    {
      type: 'item',
      href: `${config.LMS_BASE_URL}/account/settings`,
      content: 'Account',
    },
    logoutMenuItem,
  ];

  if (getConfig().MINIMAL_HEADER && authenticatedUser !== null) {
    userMenu = [
      dashboardMenuItem,
      logoutMenuItem,
    ];
  }

  const loggedOutItems = [
    {
      type: 'item',
      href: config.LOGIN_URL,
      content: 'Login',
    },
  ];

  const props = {
    logo: LogoSVG,
    logoAltText: 'edX',
    siteName: 'edX',
    logoDestination: getConfig().MINIMAL_HEADER ? null : `${config.LMS_BASE_URL}/dashboard`,
    loggedIn: authenticatedUser !== null,
    username: authenticatedUser !== null ? authenticatedUser.username : null,
    avatar: authenticatedUser !== null ? authenticatedUser.avatar : null,
    mainMenu: getConfig().MINIMAL_HEADER ? [] : mainMenu,
    userMenu,
    loggedOutItems,
  };

  return (
    <>
      <Responsive maxWidth={768}>
        <MobileHeader {...props} />
      </Responsive>
      <Responsive minWidth={769}>
        <DesktopHeader {...props} />
      </Responsive>
    </>
  );
}
