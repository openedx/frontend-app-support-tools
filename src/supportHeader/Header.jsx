import React, { useContext } from 'react';
import Responsive from 'react-responsive';
import { AppContext } from '@edx/frontend-platform/react';
import {
  APP_CONFIG_INITIALIZED,
  ensureConfig,
  mergeConfig,
  getConfig,
  subscribe,
} from '@edx/frontend-platform';

import DesktopHeader from './DesktopHeader';
import MobileHeader from './MobileHeader';
import ROUTES from '../data/constants/routes';

ensureConfig([
  'LMS_BASE_URL',
  'LOGOUT_URL',
  'LOGIN_URL',
  'MARKETING_SITE_BASE_URL',
  'ORDER_HISTORY_URL',
  'ACCOUNT_PROFILE_URL',
  'ACCOUNT_SETTINGS_URL',
], 'Header component');

subscribe(APP_CONFIG_INITIALIZED, () => {
  mergeConfig({
    MINIMAL_HEADER: !!process.env.MINIMAL_HEADER,
    PUBLISHER_BASE_URL: process.env.PUBLISHER_BASE_URL,
    SUPPORT_CONFLUENCE: process.env.SUPPORT_CONFLUENCE,
    SUPPORT_CUSTOMER_REQUEST: process.env.SUPPORT_CUSTOMER_REQUEST,
  }, 'Header additional config');
});

export default function Header() {
  const { authenticatedUser, config } = useContext(AppContext);
  let COURSES_INTERNAL = config.LMS_BASE_URL;
  let DISCOVERY_INTERNAL = config.DISCOVERY_API_BASE_URL;
  let CREDENTIALS_INTERNAL = config.CREDENTIALS_BASE_URL;
  const { SUPPORT_CONFLUENCE, SUPPORT_CUSTOMER_REQUEST } = config;
  const { CONFIGURATION } = ROUTES;

  if (config.LMS_BASE_URL.indexOf('.stage.') !== -1) {
    COURSES_INTERNAL = COURSES_INTERNAL.replace('.stage.', '-internal.stage.');
    DISCOVERY_INTERNAL = DISCOVERY_INTERNAL.replace('.stage.', '-internal.stage.');
    CREDENTIALS_INTERNAL = CREDENTIALS_INTERNAL.replace('.stage.', '-internal.stage.');
  } else if (config.LMS_BASE_URL.indexOf('.edx.') !== -1) {
    COURSES_INTERNAL = COURSES_INTERNAL.replace('.edx.', '-internal.edx.');
    DISCOVERY_INTERNAL = DISCOVERY_INTERNAL.replace('.edx.', '-internal.edx.');
    CREDENTIALS_INTERNAL = CREDENTIALS_INTERNAL.replace('.edx.', '-internal.edx.');
  }
  const mainMenu = [
    {
      type: 'item',
      content: 'Confluence',
      href: SUPPORT_CONFLUENCE,
    },
    {
      type: 'item',
      content: 'Support Tools',
      href: `${config.BASE_URL}`,
    },
    {
      type: 'submenu',
      content: 'Escalations',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href="https://edx.lightning.force.com/lightning/o/Case/list?filterName=00B0L000005D8BgUAK">Salesforce</a></div>
          <div className="mb-1"><a rel="noopener" href={SUPPORT_CUSTOMER_REQUEST}>CR Requests</a></div>
        </>
      ),
    },
    {
      type: 'submenu',
      content: 'Payment',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.ECOMMERCE_BASE_URL}/dashboard/users/`}>Otto</a></div>
          <div className="mb-1"><a rel="noopener" href="https://ebc2.cybersource.com/ebc2/app/Home">Cybersource</a></div>
          <div className="mb-1"><a rel="noopener" href="https://www.paypal.com/mep/dashboard">Paypal</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.ECOMMERCE_BASE_URL}/enterprise/coupons/`}>Enterprise Coupons</a></div>
        </>
      ),
    },
    {
      type: 'submenu',
      content: 'Courses',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.PUBLISHER_BASE_URL}`}>Publisher</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.DISCOVERY_API_BASE_URL}`}>Discovery</a></div>
          <div className="mb-1"><a rel="noopener" href={`${config.LMS_BASE_URL}/courses`}>Course Catalogue</a></div>
        </>
      ),
    },
    {
      type: 'item',
      content: 'Proctoring',
      href: `${COURSES_INTERNAL}/admin/edx_proctoring/proctoredexamsoftwaresecurereview/`,
    },
    {
      type: 'submenu',
      content: 'Programs',
      submenuContent: (
        <>
          <div className="mb-1"><a rel="noopener" href={`${config.CREDENTIALS_BASE_URL}/records/`}>Learner Record</a></div>
          <div className="mb-1"><a rel="noopener" href={`${CREDENTIALS_INTERNAL}/admin/credentials/usercredential/`}>Credentials Search</a></div>
          <div className="mb-1"><a rel="noopener" href={`${DISCOVERY_INTERNAL}/admin/course_metadata/program/`}>Programs Discovery</a></div>
        </>
      ),
    },
  ];
  /* Start Configuration Management */
  // Add configuration dropdown if FEATURE_CONFIGURATION_MANAGEMENT is set
  const configurationDropdown = {
    type: 'submenu',
    content: 'Enterprise Setup',
    submenuContent:
      getConfig().FEATURE_CONFIGURATION_ENTERPRISE_PROVISION
        ? (<div className="mb-1"><a rel="noopener" href={`${config.BASE_URL}${CONFIGURATION.SUB_DIRECTORY.PROVISIONING.HOME}`}>Learner Credit Plans</a></div>)
        : null,
  };
  if (getConfig().FEATURE_CONFIGURATION_MANAGEMENT) {
    mainMenu.push(configurationDropdown);
  }
  /* End Configuration Management */

  const dashboardMenuItem = {
    type: 'item',
    href: `${config.LMS_BASE_URL}/dashboard`,
    content: 'Dashboard',
  };

  const logoutMenuItem = {
    type: 'item',
    href: config.LOGOUT_URL,
    content: 'Logout',
  };

  let userMenu = authenticatedUser === null ? [] : [
    dashboardMenuItem,
    {
      type: 'item',
      href: `${config.ACCOUNT_PROFILE_URL}/u/${authenticatedUser.username}`,
      content: 'Profile',
    },
    {
      type: 'item',
      href: config.ACCOUNT_SETTINGS_URL,
      content: 'Account',
    },
    logoutMenuItem,
  ];

  if (getConfig().MINIMAL_HEADER && authenticatedUser !== null) {
    userMenu = [
      dashboardMenuItem,
      logoutMenuItem,
    ];
  }

  const loggedOutItems = [
    {
      type: 'item',
      href: config.LOGIN_URL,
      content: 'Login',
    },
  ];

  const props = {
    logo: config.LOGO_URL,
    logoAltText: 'edX',
    siteName: 'edX',
    logoDestination: getConfig().MINIMAL_HEADER ? null : `${config.LMS_BASE_URL}/dashboard`,
    loggedIn: authenticatedUser !== null,
    username: authenticatedUser !== null ? authenticatedUser.username : null,
    avatar: authenticatedUser !== null ? authenticatedUser.avatar : null,
    mainMenu: getConfig().MINIMAL_HEADER ? [] : mainMenu,
    userMenu,
    loggedOutItems,
  };

  return (
    <>
      <Responsive maxWidth={768}>
        <MobileHeader {...props} />
      </Responsive>
      <Responsive minWidth={769}>
        <DesktopHeader {...props} />
      </Responsive>
    </>
  );
}
