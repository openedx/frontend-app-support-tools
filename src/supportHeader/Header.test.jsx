import React from 'react';
import { mergeConfig } from '@edx/frontend-platform';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { render } from '@testing-library/react';
import { Context as ResponsiveContext } from 'react-responsive';
import { AppContext } from '@edx/frontend-platform/react';

import Header from './Header';

describe('<Header />', () => {
  const appContextValue = {
    authenticatedUser: {
      userId: '123',
      username: 'abc',
      name: 'full name',
    },
    config: {
      LMS_BASE_URL: process.env.LMS_BASE_URL,
      LOGO_URL: 'logo_url.jpg',
    },
  };

  it('renders correctly for authenticated desktop', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 1200 }}>
        <IntlProvider locale="en">
          <AppContext.Provider value={appContextValue}>
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const { container: wrapper } = render(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly for authenticated desktop when HIDE_USERNAME_FROM_HEADER is set to true', () => {
    mergeConfig({
      HIDE_USERNAME_FROM_HEADER: true,
    });

    const component = (
      <ResponsiveContext.Provider value={{ width: 1200 }}>
        <IntlProvider locale="en">
          <AppContext.Provider value={appContextValue}>
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const { container: wrapper } = render(component);
    expect(wrapper).toMatchSnapshot();

    mergeConfig({
      HIDE_USERNAME_FROM_HEADER: '',
    });
  });

  it('renders correctly for authenticated mobile', () => {
    const component = (
      <ResponsiveContext.Provider value={{ width: 500 }}>
        <IntlProvider locale="en">
          <AppContext.Provider value={appContextValue}>
            <Header />
          </AppContext.Provider>
        </IntlProvider>
      </ResponsiveContext.Provider>
    );

    const { container: wrapper } = render(component);
    expect(wrapper).toMatchSnapshot();
  });
});
