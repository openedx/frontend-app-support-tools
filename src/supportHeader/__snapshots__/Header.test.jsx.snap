// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly for authenticated desktop 1`] = `
<div>
  <header
    class="site-header-desktop"
  >
    <div
      class="container-fluid"
    >
      <div
        class="nav-container position-relative d-flex align-items-center"
      >
        <a
          class="logo"
          href="http://localhost:18000/dashboard"
        >
          <img
            alt="edX"
            class="d-block"
            src="logo_url.jpg"
          />
        </a>
        <nav
          aria-label="Main"
          class="nav main-nav"
        >
          <a
            class="nav-link"
          >
            Confluence
          </a>
          <a
            class="nav-link"
            href="undefined"
          >
            Support Tools
          </a>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Escalations
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Payment
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Courses
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <a
            class="nav-link"
            href="http://localhost:18000/admin/edx_proctoring/proctoredexamsoftwaresecurereview/"
          >
            Proctoring
          </a>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Programs
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
        </nav>
        <nav
          aria-label="Secondary"
          class="nav secondary-menu-container align-items-center ml-auto"
        >
          <div
            class="pgn__dropdown pgn__dropdown-light dropdown"
            data-testid="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md dropdown-toggle btn btn-tertiary btn-md"
              id="desktop-header-dropdown-toggle"
              type="button"
            >
              <img
                alt=""
                class="pgn__avatar pgn__avatar-sm"
                src="icon/mock/path"
              />
              abc
            </button>
          </div>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> renders correctly for authenticated desktop when HIDE_USERNAME_FROM_HEADER is set to true 1`] = `
<div>
  <header
    class="site-header-desktop"
  >
    <div
      class="container-fluid"
    >
      <div
        class="nav-container position-relative d-flex align-items-center"
      >
        <a
          class="logo"
          href="http://localhost:18000/dashboard"
        >
          <img
            alt="edX"
            class="d-block"
            src="logo_url.jpg"
          />
        </a>
        <nav
          aria-label="Main"
          class="nav main-nav"
        >
          <a
            class="nav-link"
          >
            Confluence
          </a>
          <a
            class="nav-link"
            href="undefined"
          >
            Support Tools
          </a>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Escalations
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Payment
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Courses
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
          <a
            class="nav-link"
            href="http://localhost:18000/admin/edx_proctoring/proctoredexamsoftwaresecurereview/"
          >
            Proctoring
          </a>
          <div
            class="menu nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="menu-trigger nav-link d-inline-flex align-items-center"
            >
              Programs
               
              <svg
                aria-hidden="true"
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </a>
          </div>
        </nav>
        <nav
          aria-label="Secondary"
          class="nav secondary-menu-container align-items-center ml-auto"
        >
          <div
            class="pgn__dropdown pgn__dropdown-light dropdown"
            data-testid="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md dropdown-toggle pgn__avatar-button-hide-label btn btn-tertiary btn-md"
              id="desktop-header-dropdown-toggle"
              type="button"
            >
              <img
                alt="full name"
                class="pgn__avatar pgn__avatar-sm"
                src="icon/mock/path"
              />
            </button>
          </div>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> renders correctly for authenticated mobile 1`] = `
<div>
  <header
    aria-label="Main"
    class="site-header-mobile d-flex justify-content-between align-items-center shadow sticky-top"
  >
    <div
      class="w-100 d-flex justify-content-start"
    >
      <div
        class="menu position-static"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="Main Menu"
          class="menu-trigger icon-button"
          title="Main Menu"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="24px"
            role="img"
            style="width: 1.5rem; height: 1.5rem;"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <rect
              fill="currentColor"
              height="2"
              width="20"
              x="2"
              y="5"
            />
            <rect
              fill="currentColor"
              height="2"
              width="20"
              x="2"
              y="11"
            />
            <rect
              fill="currentColor"
              height="2"
              width="20"
              x="2"
              y="17"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="w-100 d-flex justify-content-center"
    >
      <a
        class="logo"
        href="http://localhost:18000/dashboard"
        itemtype="http://schema.org/Organization"
      >
        <img
          alt="edX"
          class="d-block"
          src="logo_url.jpg"
        />
      </a>
    </div>
    <div
      class="w-100 d-flex justify-content-end align-items-center"
    >
      <div
        class="pgn__dropdown pgn__dropdown-light dropdown"
        data-testid="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-md dropdown-toggle pgn__avatar-button-hide-label btn btn-tertiary btn-md"
          id="mobile-header-dropdown-toggle"
          type="button"
        >
          <img
            alt="abc"
            class="pgn__avatar pgn__avatar-sm"
            src="icon/mock/path"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
